File: HRM_system/src/main/java/com/emranhss/HRM_system/user/UserService.java
package com.emranhss.HRM_system.user;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface UserService {
    
    UserResponseDto createUser(UserRequestDto dto);

    
    List<UserResponseDto> getAllUsers();

    
    UserResponseDto getUserById(Long id);

    
    UserResponseDto updateUser(Long id,
                               UserRequestDto dto);

    
    void deleteUser(Long id);

    
    UserResponseDto getUserByEmail(String email);

    UserResponseDto assignRole(Long id, UserRequestDto dto);

    UserResponseDto activateUser(Long id);

    UserResponseDto deactivateUser(Long id);

    void resetPassword(Long id, UserRequestDto dto);

    void changePassword(Long id, UserRequestDto dto);

}

File: HRM_systemAnguler/src/app/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../models/user.model';
import { environment } from '../../environments/environment';
@Injectable({
  providedIn: 'root',
})
export class UserService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.baseUrl}/api/user`;
  getAllUsers(): Observable<User[]> {
    return this.http.get<User[]>(this.apiUrl);
  }
  getUserById(id: number): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/${id}`);
  }
  getUserByEmail(email: string): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/email/${email}`);
  }
  createUser(user: User): Observable<User> {
    return this.http.post<User>(this.apiUrl, user);
  }
  updateUser(id: number, user: User): Observable<User> {
    return this.http.put<User>(`${this.apiUrl}/${id}`, user);
  }
  deleteUser(id: number): Observable<string> {
    return this.http.delete(`${this.apiUrl}/${id}`, {
      responseType: 'text',
    });
  }
  changePassword(id: number, oldPassword: string, newPassword: string): Observable<string> {
    return this.http.put(
      `${this.apiUrl}/${id}/change-password`,
      { oldPassword, newPassword },
      {
        responseType: 'text',
      },
    );
  }
}
